src/go/k8s/controllers/redpanda/console_controller_test.go

```diff
@@ -193,13 +193,20 @@ var _ = Describe("Console controller", func() {
 	Context("When updating Console", func() {
 		ctx := context.Background()
 		It("Should not create new ConfigMap if no change on spec", func() {
+			var configmapNsn string
+
 			By("Getting Console")
 			consoleLookupKey := types.NamespacedName{Name: ConsoleName, Namespace: ConsoleNamespace}
-			createdConsole := &redpandav1alpha1.Console{}
-			Expect(k8sClient.Get(ctx, consoleLookupKey, createdConsole)).Should(Succeed())
+			Eventually(func() bool {
+				createdConsole := &redpandav1alpha1.Console{}
+				Expect(k8sClient.Get(ctx, consoleLookupKey, createdConsole)).Should(Succeed())
 
-			ref := createdConsole.Status.ConfigMapRef
-			configmapNsn := fmt.Sprintf("%s/%s", ref.Namespace, ref.Name)
+				if ref := createdConsole.Status.ConfigMapRef; ref != nil {
+					configmapNsn = fmt.Sprintf("%s/%s", ref.Namespace, ref.Name)
+					return true
+				}
+				return false
+			}, timeout, interval).Should(BeTrue())
```

Comment:
One thing I would like to suggest is adding messages to the assertions so that someone reading failure logs can gain some understanding by simply reading the failure message:

```suggestion
			}, timeout, interval).Should(BeTrue(), "some text explaining why this failure matters: %v", someVariableThatIsRelevant)
```

Reply:
Updated